agent/test-agent-cli/design/RUNBOOK.md

```diff
@@ -0,0 +1,98 @@
+# Steps to use Test-agent Command Line Interface
+`test-agent-cli` lets you write Testsys test using Bash and help in receiving and sending information from/to the Testsys cluster.
+
+## Prerequisites:
+* kind: https://kind.sigs.k8s.io/docs/user/quick-start/
+* Kubectl: https://kubernetes.io/docs/tasks/tools/
+* Docker : https://docs.docker.com/get-started/
+
+## Steps to install Testsys
+Set the TESTSYS_DIR variable to point to the directory in which you have cloned the project. For example:
+```shell
+export TESTSYS_DIR="${HOME}/repos/bottlerocket-test-system"
+```
+Set alias   
+```shell
+alias cli="${TESTSYS_DIR}/.cargo/bin/cli"
+```
+Install the `cli` command line tool into the local CARGO_HOME as: 
+   ```shell
+   cd "${TESTSYS_DIR}"
+   cargo install --path "${TESTSYS_DIR}/cli" --force
+   ```
+
+## Steps to create Bash based TestAgent:
+
+The following commands can be used to communicate with a testsys cluster. Create a bash script like [Example test](../examples/example_test_agent_cli/example-test.sh).
+```shell
+# Get the configuration details and setting the task state running
+test-agent-cli init
+
+# Get the number of retires allowed in case of failing tests
+test-agent-cli retry-count
+
+# Get the secret value using the secret key
+test-agent-cli get-secret secret-key
+
+# Send the result of every test run to test object in Controller
+test-agent-cli send-result -o pass -p 1 -f 0 -s 0
+
+# Send any error encountered in test
+test-agent-cli send-error error-message
+
+# Mark the test as completed
+test-agent-cli terminate --results-dir results_directory
+```
+Create a [Dockerfile](../examples/example_test_agent_cli/Dockerfile).\
+Remember to set the ENTRYPOINT to the test Bash script and install the required packages.
+
+Create the docker image.
+   ```shell 
+   make example-test-agent-cli 
```

Comment:
Can you add a note that users should add a target to the makefile if they want to create their new image.